### PR TITLE
Handle missing links gracefully.

### PR DESCRIPTION
**Pull request by memory**
Herein, we do two things:

1- implement a custom handler for the virtual filesystem that,
   rather than returning a simple (and ugly) 404 page, redirects
   the client back to the root URL with the `customUrl` query
   parameter filled out with the value of the request path.

2- In home.js, if the `customUrl` param is filled out, automatically
   select the `custom` state setting, and pre-fill out the CustomID
   input field with the value of that param.

In short, the server will never again return a 404 error, but instead
will gracefully prompt the user to fill in the missing link.

### CODE REVIEW

**Comment by memory** [comment]
n.b. if there's a simpler way to capture 404 errors coming out of `net/http.FileSystem()` I wasn't able to figure it out but I'd be _delighted_ to find one.

**Comment by memory** [comment]
(Whatever's going on w/ the Travis builds here is, I think, unrelated to this PR?)

**Comment by mxschmitt** [comment]
LGTM, thanks!

**Comment by mxschmitt** [comment]
Hey @memory, sorry for the slight delay.

I've fixed the CI, it was a strange short if statement, not sure if it was from you. 
You can take a look here in this commit: 238835460bf05d7537f344eab25c8f38e5b90597

Best regards

Max

**Comment by memory** [comment]
Whoops!  That was in fact me -- I'm a total javascript/react n00b as if you couldn't tell. :)
